primitives/src/merkle_tree/mod.rs

```diff
@@ -220,15 +220,10 @@ pub trait MerkleTreeScheme: Sized {
     ) -> LookupResult<Self::Element, Self::MembershipProof, ()>;
 
     /// Verify an element is a leaf of a Merkle tree given the proof
-    /// * `pos` - zero-based index of the leaf in the tree
     /// * `proof` - a merkle tree proof
     /// * `returns` - Ok(true) if the proof is accepted, Ok(false) if not. Err()
     ///   if the proof is not well structured, E.g. not for this merkle tree.
-    fn verify(
-        &self,
-        pos: impl Borrow<Self::Index>,
-        proof: impl Borrow<Self::MembershipProof>,
-    ) -> Result<bool, PrimitivesError>;
+    fn verify(proof: impl Borrow<Self::MembershipProof>) -> Result<bool, PrimitivesError>;
```

Comment:
Should we change the return type to `Result<(), PrimitivesError>` as well?

Reply:
How about this `pub type VerificationResult = Result<(), ()>`? Just to keep the code cleaner.

Reply:
Done in https://github.com/EspressoSystems/jellyfish/pull/274/commits/37da39a17bb95b23c29c07b3bcd028069e1f2fe4, https://github.com/EspressoSystems/jellyfish/pull/274/commits/27d5e8e962271761852e5b905b1f88f9390daf78